app/assets/javascripts/fae/form/_validator.js

```diff
@@ -21,37 +24,83 @@ Fae.form.validator = {
-   * @fires {@link navigation.language.checkForHiddenErrors}
    */
   formValidate: function () {
     var _this = this;
+
     FCH.$document.on('submit', 'form', function (e) {
-      _this.is_valid = true;
+      var $this = $(this);
 
-      // Scope the data-validation only to the form submitted
-      $('[data-validate]', $(this)).each(function () {
-        if ($(this).data('validate').length) {
-          _this._judgeIt($(this));
-        }
-      });
+      if ($this.data('passed_validation') !== 'true') {
 
-      // Catch visible errors for image/file inputs hitting the fae config file size limiter
-      $('.input.file', $(this)).each(function () {
-        if ($(this).hasClass('field_with_errors')) {
-          _this.is_valid = false
+        // pause form submission
+        e.preventDefault();
+
+        // set defaults
+        _this.is_valid = true;
+        _this.validations_called = 0;
+        _this.validations_returned = 0;
+        _this.validation_test_count = 0;
+
+        // Scope the data-validation only to the form submitted
+        $('[data-validate]', $(this)).each(function () {
+          if ($(this).data('validate').length) {
+            _this.validations_called++;
+            _this._judgeIt($(this));
+          }
+        });
+
+        // Catch visible errors for image/file inputs hitting the fae config file size limiter
+        $('.input.file', $(this)).each(function () {
+          if ($(this).hasClass('field_with_errors')) {
+            _this.is_valid = false;
+          }
+        });
+
+        _this.testValidation($this);
+
+      };
+
+    });
+  },
+
+  /**
+   * Tests a forms validation after all validation checks have responded
+   * Polls validations responses every 50ms to allow uniqueness AJAX calls to complete
+   */
+  testValidation: function($this) {
+    var _this = this;
+    _this.validation_test_count++;
+
+    setTimeout(function(){
+
+      // if all the validation checks have returned a response
+      if (_this.validations_called === _this.validations_returned) {
+
+        if (_this.is_valid) {
+          // if form is valid, submit it
+          $this.data('passed_validation', 'true');
+          $this.submit();
+        } else {
+          // otherwise scroll to the top to display alerts
+          Fae.navigation.language.checkForHiddenErrors();
+          FCH.smoothScroll($('#js-main-header'), 500, 100, 0);
+
+          if ($(".field_with_errors").length) {
+            $('.alert').slideDown('fast').delay(3000).slideUp('fast');
+          }
         }
-      });
 
-      if (_this.is_valid === false) {
-        Fae.navigation.language.checkForHiddenErrors();
-        FCH.smoothScroll($('#js-main-header'), 500, 100, 0);
-        e.preventDefault();
+      } else {
+        // check again if it hasn't run more than 50 times
+        // (to prevent against infinite loop)
+        if (_this.validation_test_count < 50) {
```

Comment:
This feels uncomfortably hacky. I'm assuming the `50` was arbitrarily selected, but setting a timeout of `2500` would also be an arbitrary number. However, since it is based on AJAX that may or may not send, it's the best solution. 

Ultimately, the ideal solution would be to receive a Promise (or equivalent, you're more familiar with Judge's internals than I am) from Judge that resolves after all validations have returned from the server.

Reply:
Yep 50 is an arbitrary number, but it is merely a fallback to prevent an infinite loop.

I looked into a callback or promise on Judge's side, but I feel it would take a lot of supporting code to pass it to Fae. The AJAX can is written in vanilla JS and the call itself, where I'd usually receive the promise is in Judge as well.